### ISSUE TITLE
Add Dune 2000 campaign

### ISSUE THREAD

**Issue by penev92** [OpenRA/OpenRA#9287]
This ticket is meant to track the development of the campaigns for the Dune 2000 mod.
### Atreides campaign

| Done? | Merged | Mission | Original name | Assignee | PR |
| --- | --- | --- | --- | --- | --- |
| ✓ | ✓ | Atreides01a | A1V1 | @abcdefg30 | #8117 |
| ✓ | ✓ | Atreides01b | A1V2 | @abcdefg30 | #8157 |
| ✓ | ✓ | Atreides02a | A2V1 | @abcdefg30 | #9402 |
| ✓ | ✓ | Atreides02b | A2V2 | @abcdefg30 | #9512 |
| ✓ | ✓ | Atreides03a | A3V1 | @abcdefg30 | #9659 |
| ✓ | ✓ | Atreides03b | A3V2 | @abcdefg30 | #11706 |
| ✓ | ✓ | Atreides04 | A4V1 | @abcdefg30 | #12743 |
| ✓ | ✓ | Atreides05 | A5V1 | @abcdefg30 | #12785 |
|  |  | Atreides06a | A6V1 | @abcdefg30 |  |
|  |  | Atreides06b | A6V2 | @abcdefg30 |  |
|  |  | Atreides07 | A7V1 | @abcdefg30 |  |
|  |  | Atreides08 | A8V1 | @abcdefg30 |  |
|  |  | Atreides09a | A9V1 | @abcdefg30 |  |
|  |  | Atreides09b | A9V2 | @abcdefg30 |  |
### Harkonnen campaign

| Done? | Merged | Mission | Original name | Assignee | PR |
| --- | --- | --- | --- | --- | --- |
| ✓ | ✓ | Harkonnen01a | H1V1 | @MustaphaTR | #12755 |
| ✓ | ✓ | Harkonnen01b | H1V2 | @MustaphaTR | #12779 |
| ✓ | ✓ | Harkonnen02a | H2V1 | @MustaphaTR | #12811 |
| ✓ | ✓ | Harkonnen02b | H2V2 | @MustaphaTR | #12832 |
| ✓ | ✓ | Harkonnen03a | H3V1 | @MustaphaTR | #13118 |
| ✓ | ✓ | Harkonnen03b | H3V2 | @MustaphaTR | #13156 |
| ✓ | ✓ | Harkonnen04 | H4V1 | @MustaphaTR | #13377 |
| ✓ | ✓ | Harkonnen05 | H5V1 | @MustaphaTR | #14388 |
| ✓ | ✓ | Harkonnen06a | H6V1 | @MustaphaTR | #14164 |
| ✓ | ✓ | Harkonnen06b | H6V2 | @MustaphaTR |  #14288 |
| ✓ | ✓ | Harkonnen07 | H7V1 | @MustaphaTR | #14426 |
| ✓ | ✓ | Harkonnen08 | H8V1 | @MustaphaTR | #14427 |
| ✓ | ✓ | Harkonnen09a | H9V1 | @MustaphaTR | #14429 |
| ✓ | ✓ | Harkonnen09b | H9V2 | @MustaphaTR | #14434 |
### Ordos campaign

| Done? | Merged | Mission | Original name | Assignee | PR |
| --- | --- | --- | --- | --- | --- |
| ✓ | ✓ | Ordos01a | O1V1 |  @MustaphaTR | #12664 |
| ✓ | ✓ | Ordos01b | O1V2 | @MustaphaTR  | #12732 |
| ✓ | ✓ | Ordos02a | O2V1 | @MustaphaTR | #12880 |
| ✓ | ✓ | Ordos02b | O2V2 | @MustaphaTR | #13047 |
| ✓ | ✓ | Ordos03a | O3V1 | @MustaphaTR | #13259 |
| ✓ | ✓ | Ordos03b | O3V2 | @MustaphaTR | #13324 |
| ✓ | ✓ | Ordos04 | O4V1 | @MustaphaTR | #13063 |
| ✓ | ✓ | Ordos05 | O5V1 | @MustaphaTR  | #14798 |
| ✓ | ✓ | Ordos06a | O6V1 | @abcdefg30  | #17580 |
| ✓ |  | Ordos06b | O6V2 | @MustaphaTR  |  |
|  |  | Ordos07 | O7V1 |  |  |
|  |  | Ordos08 | O8V1 |  |  |
|  |  | Ordos09a | O9V1 |  |  |
|  |  | Ordos09b | O9V2 |  |  |

If you want to port a mission, please leave a comment below.

**Comment by CH4Code** [comment]
I'm thinking about tackling the ordos missions and wonder about the bounty thing, which seems interesting. But how does it work? Is the bounty paid on completion of individiual missions, on the completion of full campaigns or only if the whole ticket is completed?

**Comment by pchote** [comment]
Bounties can only be claimed when a ticket is completely closed.  This is why we discourage people from placing bounties on big tickets like this (which have little hope of being finished by one person), but so far our request for a technical solution to the problem (https://github.com/bountysource/core/issues/968) has been ignored.

**Comment by CH4Code** [comment]
I thought so... :/   
Heard of this issue previously in another github Group.

**Comment by tomasz89** [comment]
I was watching this ticket thinking to contribute, but not because the bounty interests me. Maybe this is another approach - the participants could nominate whether they're interested in sharing in the bounty when considering contributing?

**Comment by pchote** [comment]
We could replace these meta tickets with the new github projects.

**Comment by CH4Code** [comment]
Hey, 
I converted all the campaign maps (without script), using correct tilesets (some are flawless, some still have graphical errors) and put them [here ](https://github.com/CH4Code/D2k-Campaign-Resources) for all to access and use. Please put the link into the ticket description.

I will do no lua scripting, so someone else can work on this.

If I find the bug(s) in the map converter, I will upload the flawless maps later.

You can also find savegames of all D2k missions there!

**Comment by CH4Code** [comment]
I fixed the converter bug. All 39 maps on my repository are now flawless :)

**Comment by MustaphaTR** [comment]
I'm working on Ordos01a and b.

**Comment by MustaphaTR** [comment]
I started to make Atr4 too.

**Comment by abcdefg30** [comment]
> I started to make Atr4 too.

Please don't. I already have that finished. I just need to sort out why harkonnen troops are unable to attack the `sietch` actor.

**Comment by MustaphaTR** [comment]
Ok, but i don't remember they did in actual game tho. Maybe only the reinforcements did, not sure. They are not Owned by Harkonnen but Ordos (but still use harkonnen palette) i think.

**Comment by MustaphaTR** [comment]
I already made both 1st and 2nd Harkonnen maps.

**Comment by CH4Code** [comment]
Gaining some momentum here :) 👍  [Here](https://github.com/OpenRA/OpenRA/wiki/Mission-Map-Tilesets) you can find a list of other maps of the campaign which should be scriptable without repairing tileset errors.

**Comment by MustaphaTR** [comment]
Harkonnen 2b is now merged and i also made Harkkonnen 3a already, can someone update the list.

**Comment by MustaphaTR** [comment]
I already completed Harkonnen 3b and Ordos 3a too. Ordos 3b is currently WIP and i'll possibly do Harkonnen 4 too after that.

**Comment by MustaphaTR** [comment]
Har4 is added, i made Har6a and Har6b today.

**Comment by MustaphaTR** [comment]
Sorry for spamming there, but i completed the Harkonnen campaign. All we need to do is get them merged now.

**Comment by penev92** [comment]
All remaining Harkonnen missions are now PR-ed. Calling @ltem and @abc013 to do some reviewing on those ;)

**Comment by pchote** [comment]
An important note to the people who placed a bounty on this issue: Bountysource have just changed their terms of service to take it for themselves!

> If no Solution is accepted within two years after a Bounty is posted, then the Bounty will be withdrawn and the amount posted for the Bounty will be retained by Bountysource. For Bounties posted before June 30, 2018, the Backer may redeploy their Bounty to a new Issue by contacting support@bountysource.com before July 1, 2020. If the Backer does not redeploy their Bounty by the deadline, the Bounty will be withdrawn and the amount posted for the Bounty will be retained by Bountysource.

I suggest creating an issue(s) for your favorite mission(s) and asking bountysource to redeploy the bounty onto it. This would provide a more immediate benefit to the people working hard to port the remaining missions, rather than allowing Bountysource to take it for themselves.

**Comment by TheSprony** [comment]
I have to be real with you, I'm no programmer and I don't use Github. I just made this account to ask if there's some way to finance this so it gets completed? If so, how much would that cost? I want to be able to play this in OpenRA and I've been waiting for a long time to see any progress, but that's not happening at all.

**Comment by nekromoff** [comment]
Patreon etc.?

**Comment by TheSprony** [comment]
> Patreon etc.?

I'm down with whatever gets these Dune missions into OpenRA.

**Comment by nekromoff** [comment]
Is there some manual/tips somewhere? I could volunteer a bit.

**Comment by TheSprony** [comment]
If I can help with anything, I'm down for that of course.

**Comment by abcdefg30** [comment]
> Is there some manual/tips somewhere?

Sadly not really, just the existing missions. Currently most progress is more or less blocked by rewriting our AI to support base building properly, which isn't an easy task for people unfamiliar with the engine to work on.

**Comment by TheSprony** [comment]
Are there any updates perhaps?

**Comment by abcdefg30** [comment]
No, sorry.

**Comment by TheSprony** [comment]
> No, sorry.

Thank you for the quick reply. I'll check back in another 6 months or so.

**Comment by DW76** [comment]
Apologies if this has been asked and answered, but is there a way to view the introduction cut scenes of Dune 2000? In other words, run the game where there is the Intro, House selection, etc. Thanks!

**Comment by penev92** [comment]
Unfortunately no, which is actually a big missed opportunity.

**Comment by jorgefm87** [comment]
Hello, is there any update or forecast for completing the remaining missions? It would be great if this could be finished someday.

Anyway, congratulations on the work done.

**Comment by Porenutak** [comment]
I will try to finish Atreides campaign this year.

**Comment by Porenutak** [comment]
Missions are almost complete, but there is still some features that are missing compared to OG
 Just for recording, here is my personal TODO list:
-	Allow AI to repair vehicles: This will require reworking attack logic and add option to issue repair command via LUA.
-	Allow AI to use Starport.
-	Implement morale system: according to d2k editor AI got some sort of morale system that I guess affects attack decision and squad retreat.
-	Add tutorial that will introduce ORA D2k unique features: Learn player that he can reclaim husk with engineer. Give him hit where to place refinery for most effective mining. Mission 1 is best for this.
-  Production via LUA get stuck when producer is killed during building proces or AI lost prerequisites. #10617

**Comment by PunkPun** [comment]
With AI repair we could try to get #21052 going first. I'm sure @dnqbob would appreciate it

**Comment by JovialFeline** [comment]
>     * Production via LUA get stuck when producer is killed or AI lost prerequisites. (there was a issue about this bug but I cannot find it)

https://github.com/OpenRA/OpenRA/issues/10617 ?
It can also halt if the relevant building is destroyed while the queue is active. This doesn't seem to be a problem when the queue is contained in the building itself (CnC) but it is otherwise (RA, D2k).

**Comment by PunkPun** [comment]
> Add tutorial that will introduce ORA D2k unique features: Learn player that he can reclaim husk with engineer. Give him hit where to place refinery for most effective mining. Mission 1 is best for this.

In RA mechanics can also do it. #14763 #17238. II also believe Combined Arms has implemented such features

**Comment by CoachGodzup** [comment]
I want to contribute on this guide. How can I see the mission in Dune 2000, in order to copy this situation into OpenRA?

**Comment by PunkPun** [comment]
After 11 years, all missions are in the game 🎊

**Comment by rea987** [comment]
Will these be available in the next OpenRA release or already available?

**Comment by TheSprony** [comment]
Yeah, where are the maps, and how can I play them?

**Comment by Porenutak** [comment]
They will be available in upcoming playtest.

**Comment by TheSprony** [comment]
Has a date for that been announced yet?
